sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/util/ReduceFnRunner.java

```diff
@@ -326,6 +326,15 @@ private void collectAndMergeWindows(Iterable<WindowedValue<InputT>> values) thro
         @SuppressWarnings("unchecked")
         W window = (W) untypedWindow;
 
+        ReduceFn<K, InputT, OutputT, W>.Context directContext =
+            contextFactory.base(window, StateStyle.DIRECT);
+        if (triggerRunner.isClosed(directContext.state())) {
```

Comment:
If not merging: We will now reject here instead of next itr.
If merging: We mimic the same test in collectAndMergeWindows. The window (should not be) active.